src/operator/context/account.tsx

```diff
@@ -0,0 +1,138 @@
+// Libraries
+import React, {FC, useCallback, useEffect, useState} from 'react'
+import {useParams, useHistory} from 'react-router-dom'
+import {useDispatch} from 'react-redux'
+
+// Utils
+import {
+  getAccountById,
+  deleteAccountById,
+  removeUserFromAccount,
+} from 'src/operator/api'
+import {notify} from 'src/shared/actions/notifications'
+import {
+  getAccountError,
+  deleteAccountError,
+} from 'src/shared/copy/notifications'
+
+// Types
+import {Account} from 'src/types/operator'
+import {RemoteDataState} from 'src/types'
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface AccountContextType {
+  account: Account
+  accountStatus: RemoteDataState
+  deleteStatus: RemoteDataState
+  handleDeleteAccount: () => void
+  handleGetAccount: () => void
+  handleRemoveUserFromAccount: (id: string) => void
+  setVisible: (vis: boolean) => void
+  visible: boolean
+}
+
+export const DEFAULT_CONTEXT: AccountContextType = {
+  account: null,
+  accountStatus: RemoteDataState.NotStarted,
+  deleteStatus: RemoteDataState.NotStarted,
+  handleDeleteAccount: () => {},
+  handleGetAccount: () => {},
+  handleRemoveUserFromAccount: (_: string) => {},
+  setVisible: (_: boolean) => {},
+  visible: false,
+}
+
+export const AccountContext = React.createContext<AccountContextType>(
+  DEFAULT_CONTEXT
+)
+
+export const AccountProvider: FC<Props> = React.memo(({children}) => {
+  const [account, setAccount] = useState(null)
+  const [visible, setVisible] = useState(false)
+  const [accountStatus, setAccountStatus] = useState(RemoteDataState.NotStarted)
+  const [deleteStatus, setDeleteStatus] = useState(RemoteDataState.NotStarted)
+
+  const {accountID} = useParams()
+  const history = useHistory()
+  const dispatch = useDispatch()
+
+  const handleGetAccount = useCallback(async () => {
+    try {
+      setAccountStatus(RemoteDataState.Loading)
+      const resp = await getAccountById(accountID)
+
+      if (resp.status !== 200) {
+        setAccountStatus(RemoteDataState.Error)
+        throw new Error(resp.data.message)
+      }
+
+      setAccount(resp.data)
+      setAccountStatus(RemoteDataState.Done)
+    } catch (error) {
+      setAccountStatus(RemoteDataState.Error)
+      console.error({error})
+      dispatch(notify(getAccountError(accountID)))
```

Comment:
Interesting! So on a non-200 response, we throw an Error, but here we eat the error and dispatch a notification instead?  I'm sure this is a standard pattern in the UI codebase, but I'm curious how those two things are handled differently. If it's a long explanation, don't worry about it. This is mainly my curiosity here.

Reply:
Nah, not a long explanation. I'm not sure how the Quartz UI was handling errors that are thrown, but for the UI the page will error and give the user an option to submit an issue to the UI repo. Since that isn't always particularly helpful (especially for an internal tool), i figured a notification of an error would give the user a better way of understanding what exactly went wrong and a potentially actionable step without effectively crashing the browser